<!-- Reference:
https://answers.ros.org/question/41433/multiple-robots-simulation-and-navigation/ -->

<launch>

  <arg name="robot_number" default="0" />
  <arg name="robot_name" default="robot_$(arg robot_number)" />
  <arg name="pose" default="-x 0 -y 0 -z 0"/>
  <arg name="model" default="$(find vss_simulation)/urdf/vss_robot.xacro"/>

  <param name="robot_description"
    command="$(find xacro)/xacro.py $(arg model) robot_number:=$(arg robot_number)" />

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
        args="$(arg pose) -urdf -model $(arg robot_name) -param robot_description"
        respawn="false" output="screen" />

  <!-- Start state publishing node -->
  <node pkg="robot_state_publisher" type="robot_state_publisher"  name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>

  <!-- Load configs to publish the state of all joints -->
  <rosparam command="load"
            file="$(find vss_simulation)/config/joints.yaml"
            ns="vss_robot_joint_state_controller" />

  <!-- Load configs to control robot motors -->
  <rosparam command="load"
          file="$(find vss_simulation)/config/diffdrive.yaml"
          ns="vss_robot_diff_drive_controller" />

  <node name="vss_robot_controller_spawner" pkg="controller_manager" type="spawner"
    args="vss_robot_joint_state_controller
          vss_robot_diff_drive_controller "/>

</launch>